jazz_services-handler/index.js

```diff
@@ -22,605 +22,283 @@ This Handler looks for Service Creation events and updates Service Catalog
-																"message": "updated service "  + payload.SERVICE_NAME.S + " in service catalog."
-															});
-														}
-													});
-
-												}
-
-											});
-										}
-									}
+		rejectUnauthorized: false,
+		transform: function (body, response, resolveWithFullResponse) {
+			return response;
+		}
+	};
+};
 
-								} else {
-									//push un-interesting event to processed queue
-									processedEvents.push({
-										"sequence_id": sequenceNumber,
-										"failure_code": null,
-										"failure_message": null
-									});
-									innerCallback(null,processedEvents)
-								}
-							}catch(e){
-								HandleFailedEvents(failureCodes.PR_ERROR_2.type, JSON.stringify(e), results.payload, failureCodes.PR_ERROR_2.code);
-								innerCallback(e);
-							}
-						}]
-					}, function (err, results) {
-							callback(err);
-					});
-				}, function (err) {
-					if(err){
-						mainCallback(err)
-					} else{
-						mainCallback(null, {
-							"processed_events": processedEvents.length,
-							"failed_events": failedEvents.length
-						});
-					}
-					
-					
-				});
-			}catch(e){
-				mainCallback(e);
-			}
+var validateAuthToken = function (result) {
+	return new Promise((resolve, reject) => {
+		if (result.statusCode === 200 && result.body && result.body.data) {
+			return resolve(result.body.data.token);
+		} else {
+			return reject(errorHandler.throwUnauthorizedError("User is not authorized to access this service"));
 		}
-	}, function (err, results) {
-		logger.error("### Error : " + JSON.stringify(err));
-		if (err) {
-			cb(err);
+	});
+}
+
+var handleError = function (errorType, message) {
+	var error = {};
+	error.failure_code = errorType;
+	error.failure_message = message;
+	return error;
+}
+
+var processRecords = function (event, configData, authToken) {
+	return new Promise((resolve, reject) => {
+		var processRecordPromises = [];
+		for (i = 0; i < event.Records.length; i++) {
+			processRecordPromises.push(processRecord(event.Records[i], configData, authToken));
+		}
+		Promise.all(processRecordPromises)
+			.then((result) => { return resolve(result); })
+			.catch((error) => { return reject(error); });
+	});
+}
+
+var processRecord = function (record, configData,authToken) {
+	return new Promise((resolve, reject) => {
+		var sequenceNumber = record.kinesis.sequenceNumber;
+		var encodedPayload = record.kinesis.data;
+		var payload;
+		return checkInterest(encodedPayload, sequenceNumber)
+			.then(result => {
+				payload = result.payload;
+				if (result.interested_event) {
+					return processEvent(payload, configData, authToken);
+				} else {
+					handleProcessedEvents(sequenceNumber, payload);
+				}
+			})
+			.then(result => {
+				handleProcessedEvents(sequenceNumber, payload);
+				return resolve(result);
+			})
+			.catch(err => {
+				handleFailedEvents(sequenceNumber, err.failure_message, payload, err.failure_code);
+				return reject(err);
```

Comment:
Since we are handling failedEvent, I don't believe we need a `reject(err)` and instead should be `resolve`

Reply:
In internal this is handled in callback error. So I think the flow is correct and having parity with internal.

Reply:
IMO, it is wrong. @deepusundar, @Deepak-Babu, @devsatishm: Can you guys comment?

Reply:
Discussed with @devsatishm and @deepusundar and we agreed that we shouldn't reject (even though the code earlier was that way). Lets do the right thing here and fix it.

Reply:
This wasn't fixed and code was merged. Please create another PR with the changes requested in this PR.